File: Lab2/Python/content_worker.py
from furniture import Furniture


def select_available_types(furniture_list: list[Furniture], category: str, kind: str) -> list[Furniture]:
    return [furn for furn in furniture_list if furn.name == category and furn.kind == kind and furn.amount > 0]


def without_range(furniture_list: list[Furniture], min_value: int, max_value: int) -> list[Furniture]:
    return [furn for furn in furniture_list if min_value < furn.price > max_value ]

File: Lab2/Sharp/Lab2/Lab2/ContentWorker.cs
using System.Linq;

namespace Lab2
{
    internal static class ContentWorker
    {
        public static Furniture[] GetAvailableTypes(Furniture[] furnitures, string category, string kind)
        {
            return furnitures.Any(f => f.Name == category)
                ? furnitures.Where(f => f.Name == category && f.Kind == kind && f.Amount > 0).ToArray()
                : new Furniture[0];
        }
        public static Furniture[] GetItemsOutOfRange(Furniture[] furnitures, int minValue, int maxValue)
        {
            return furnitures.Where(f => f.Price <= minValue || f.Price >= maxValue).ToArray();
        }
    }
}
